### PR TITLE
Handle native context in setUrl

### PR DESCRIPTION
**Pull request by sa-spag**
In order to be able to open deeplinks on iOS, I implemented a solution using `xcrun simctl openurl` command.

### CODE REVIEW

File: lib/commands/general.js [hunk 1]
```diff
@@ -188,6 +188,19 @@ commands.getStrings = async function (language, stringFile = null) {
   return strings;
 };
 
+commands.setUrl = async function (url) {
+  log.debug(`Attempting to set url '${url}'`);
+  if (!this.isWebContext()) {
+    // use xcrun to open deeplink
+    await exec('xcrun', ['simctl', 'openurl', this.opts.udid || this.sim.udid, url]);
```
Comment (imurchie) [inline lib/commands/general.js]: There is a method on `node-simctl` for this command: https://github.com/appium/node-simctl/blob/master/lib/simctl.js#L67

**Comment by sa-spag** [comment]
I guess we need to create a deeplink on https://github.com/appium/ios-test-app (something like `testapp://sum?a=(integer)&b=(integer)`) to test that.

**Comment by jlipps** [comment]
@sa-spag yeah this is great. do you know how to create a deeplink scheme? would you mind doing it for TestApp?

**Comment by sa-spag** [comment]
@jlipps Yep, that's why I updated it using a build created from https://github.com/appium/ios-test-app/pull/16. Perhaps we could add https://github.com/appium/ios-test-app as a dependency and change the path [here](https://github.com/appium/appium-ios-driver/blob/master/test/e2e/testapp/desired.js#L6)?

**Comment by jlipps** [comment]
@sa-spag I think @imurchie made it this way because we were having trouble building the test app on CI, and so including it pre-built was easier. anyway, for now you've got the change here as well which is what's important. we can figure out what else to do later.

**Comment by imurchie** [comment]
We can't just have the test app as a dev dependency because we need to run this repo in Travis using Xcope 7.3 and a recent change to the test app makes it impossible to build below 8.

**Comment by imurchie** [comment]
Published in `appium-ios-driver@1.19.0`.

**Pull request merged**